lib/common/socket/evloop.c.h

```diff
@@ -535,6 +535,37 @@ static void run_pending(h2o_evloop_t *loop)
     }
 }
 
+void h2o_evloop_destroy(h2o_evloop_t *loop)
+{
+    struct st_h2o_evloop_socket_t *sock;
+    //dispose all socket
+    while (loop->_pending_as_server != NULL || loop->_pending_as_client != NULL) {
+        while ((sock = loop->_pending_as_client) != NULL) {
+            loop->_pending_as_client = sock->_next_pending;
+            sock->_next_pending = sock;
+            do_dispose_socket((h2o_socket_t *)sock);
```

Comment:
We should call `dispose_socket` (not `do_dispose_socket`). The difference is that `dispose_socket` disposes of all the resources associated to a socket. `do_dispose_socket` only disposes of the resources related to evloop.